Source/JavaScriptCore/runtime/JSFunction.cpp

```diff
@@ -466,10 +466,15 @@ CallData JSFunction::getConstructData(JSCell* cell)
 
     JSFunction* thisObject = jsCast<JSFunction*>(cell);
     if (thisObject->isHostFunction()) {
-        if (thisObject->nativeConstructor() != callHostFunctionAsConstructor) {
+        if (thisObject->inherits<JSBoundFunction>()) {
+            if (jsCast<JSBoundFunction*>(thisObject)->canConstruct()) {
+                constructData.type = CallData::Type::Native;
+                constructData.native.function = thisObject->nativeConstructor();
+                constructData.native.isBoundFunction = true;
+            }
```

Comment:
What happens in the `! canConstruct` case?  It looks like `CallData` will be uninitialized.

Reply:
Just all fields are initial value. (no change in this patch)